core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -91,7 +90,12 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   /** List all the files currently stored on disk by the disk manager. */
   def getAllFiles(): Seq[File] = {
     // Get all the files inside the array of array of directories
-    subDirs.flatten.filter(_ != null).flatMap { dir =>
+    subDirs.flatMap { dir =>
```

Comment:
In other words, due to `dir.synchronized`, if we copy a item from `dir` to the new cloned array, it will either be `null`, or a completely initialized file. Then we can safely use these files.